website/src/Pages/EventsPage/EventsPage.jsx

```diff
@@ -1,10 +1,249 @@
+    const response = await axios.get('http://localhost:4000/event/getevents', {
+      headers: authHeader,
+    });
+    const events = response.data;
+    setAllEvents(events);
+
+    const initialEvents = await identifyEventsInRange(dayjs(), 0, events);
+    setEventsInRange(initialEvents);
+  };
+
+  useEffect(() => {
+    getEvents();
+  }, [refresh]);
+
+  const updateWeek = async (newDiff) => {
+    const newWeek = dayjs().add(newDiff, 'week');
+    const updatedEvents = await identifyEventsInRange(newWeek, newDiff, allEvents);
+    setEventsInRange(updatedEvents);
+  };
+
+  const addNewEvent = () => {
+    navigate('/events/add');
+  };
+
+  const editEventDetails = (id) => {
+    const eventDetails = allEvents.find((e) => e._id === id);
+    navigate('/events/edit', {
+      state: {
+        eventDetails,
+      },
+    });
+  };
+
+  const deleteEvent = async (eventId) => {
+    await axios.delete(`http://localhost:4000/event/delete/${eventId}`, {
+      headers: authHeader,
+    });
+    setOpenPreview(false);
+    setRefresh((val) => val + 1);
+  };
+
+  const previewEventDetails = (eventDetails) => {
+    setOpenPreview(true);
+    setPreviewDetails(eventDetails);
+  };
+
   return (
-    <div>
-      <header>Events Page</header>
-    </div>
+    <Container>
+      <Box>
+        <Typography variant="h5">Events</Typography>
+      </Box>
+      <Box>
+        <Box>
+          <Button onClick={() => updateWeek(dateRange.diff - 1)}>Back</Button>
+          <Box>
+            { dateRange.mondayMonth === dateRange.sundayMonth ? (
+              <Typography>
+                {dateRange.mondayMonth}
+                {' '}
+                {dateRange.monday}
+                -
+                {dateRange.sunday}
+              </Typography>
+            ) : (
+              <Typography>
+                {dateRange.mondayMonth}
+                {' '}
+                {dateRange.monday}
+                -
+                {dateRange.sundayMonth}
+                {' '}
+                {dateRange.sunday}
+              </Typography>
+            )}
+          </Box>
+          <Button onClick={() => updateWeek(dateRange.diff + 1)}>Next</Button>
+        </Box>
+        <Box>
+          <Button
+            onClick={() => addNewEvent()}
+          >
+            Add New +
+          </Button>
+        </Box>
+      </Box>
+      <Box>
+        {eventsInRange.map((event) => (
+          <Box
+            key={event._id}
+            onClick={() => previewEventDetails(event)}
+          >
+            <EventCard
+              recurring={event.dateDetails.recurring}
+              day={event.dateDetails.day}
+              date={event.dateDetails.date}
+              startTime={event.dateDetails.startTime}
+              endTime={event.dateDetails.endTime}
+              title={event.title}
+              host={event.host}
+              tag={event.tag}
+            />
+          </Box>
+        ))}
+      </Box>
+      <Modal
```

Comment:
<img width="1440" alt="image" src="https://github.com/lablueprint/skid-row-arts-alliance/assets/53063086/34b21ad7-f01b-4d4f-aedc-72459a65b0d6">
styling is a bit off but idk if that's a matter for this PR